lib/showcase_generator.dart

```diff
@@ -62,30 +62,34 @@ class _ShowcaseGenerator extends Generator {
     }, orElse: () => null);
     final bool hasAnyRequiredParameter = _firstRequiredParameter != null;
 
-    final ConstructorElement forDesignTime = element.constructors.singleWhere(
-      (ConstructorElement c) => c.name == 'forDesignTime',
-      orElse: () => null,
-    );
+    final MethodElement forDesignTime =
+        element.lookUpMethod('forDesignTime', library.element);
     final bool isForDesignTimeDefined = forDesignTime != null;
 
     if (hasAnyRequiredParameter && !isForDesignTimeDefined) {
       throw Exception('''
 
 ${element.name} default constructor has required parameters which are not set.
-Give them default values or create a [forDesignTime] factory with default (dev-only) values.
-See https://github.com/flutter/flutter-intellij/wiki/Using-live-preview
+Give them default values or create a [forDesignTime] class method with default
+(dev-only) values, that return either [Widget] or [List<Widget>].
 
 ''');
     }
 
-    final String constructor =
-        isForDesignTimeDefined ? '${element.name}.forDesignTime' : element.name;
-
     final double boundaryWidth =
         annotatedElement.annotation.peek('width').doubleValue;
     final double boundaryHeight =
         annotatedElement.annotation.peek('height').doubleValue;
 
+    String constructor = '[${element.name}()]';
+    if (isForDesignTimeDefined &&
+        forDesignTime.returnType.displayName == 'Widget') {
```

Comment:
Is there a better way to type-check this? I'm thinking that returning a concrete `Widget` subclass from the method will make this check fail, but there really is no reason for it.

Is there a good API to check if the class can be casted to or inherits from another one?

Reply:
I'm not following, Widget seems right here, why do you want to be more specific?

Reply:
If I do:
```
static MyWidgetWithRequiredProps forDesignTime() =>
       MyWidgetWithRequiredProps('Default test-only label.');
```

instead of:
```
static Widget forDesignTime() =>
       MyWidgetWithRequiredProps('Default test-only label.');
```


This check will fail, right?

Reply:
Yes, you're right. But we have `element`: if we discover how to check the typing right, we can use it to check.